backend/LexBoxApi/GraphQL/ProjectMutations.cs

```diff
@@ -87,6 +96,64 @@ public record CreateProjectResponse(Guid? Id, CreateProjectResult Result);
         return dbContext.Projects.Where(p => p.Id == input.ProjectId);
     }
 
+    [Error<NotFoundException>]
+    [Error<DbError>]
+    [AdminRequired]
+    [UseMutationConvention]
+    public async Task<BulkAddProjectMembersResult> BulkAddProjectMembers(
+        LoggedInContext loggedInContext,
+        BulkAddProjectMembersInput input,
+        LexBoxDbContext dbContext)
+    {
+        var admin = await dbContext.Users.FindAsync(loggedInContext.User.Id);
+        var project = await dbContext.Projects.FindAsync(input.ProjectId);
+        if (project is null) throw new NotFoundException("Project not found");
+        List<string> usernameConflicts = [];
+        int createdCount = 0;
+        foreach (var username in input.Usernames)
+        {
+            var user = await dbContext.Users.Where(u => u.Username == username).FirstOrDefaultAsync();
```

Comment:
We're doing a query inside a for each, we should try to move that out so instead we're doing something like `dbContext.Users.Include(u => u.Projects).Where(u => input.Usernames.Contains(u.Username) || input.Usernames.Contains(u.Email)).ToArrayAsync();` we also need to filter on user email also, otherwise we might try creating a user that already exists just we didn't find them because we didn't filter on email.

Reply:
Done in commit 9428ec5f.